File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/dagger/demo/details/HeroDetailsModule.java
package com.github.alexeygorovoy.moxytemplate.dagger.demo.details;

import com.github.alexeygorovoy.moxytemplate.ui.demo.details.HeroDetailsPresenter;

import dagger.Module;
import dagger.Provides;


@Module
public class HeroDetailsModule {

    @Provides
    HeroDetailsPresenter heroDetailsPresenter() {
        return new HeroDetailsPresenter();
    }

}

File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/dagger/demo/details/HeroDetailsModule.kt
package com.github.alexeygorovoy.moxytemplate.dagger.demo.details

import com.github.alexeygorovoy.moxytemplate.api.models.Hero
import dagger.Module
import dagger.Provides

@Module
class HeroDetailsModule(private val hero: Hero) {

    @Provides
    internal fun provideHero(): Hero = hero
}
